android/java/org/chromium/chrome/browser/crypto_wallet/util/PortfolioHelper.java

```diff
@@ -102,23 +109,96 @@ public Double getMostRecentFiatSum() {
         return Double.valueOf(mFiatHistory[mFiatHistory.length - 1].price);
     }
 
+    public void fetchAllAssetsAndDetails(Callbacks.Callback1<PortfolioHelper> callback) {
+        resetResultData();
+        if (mActivity.get() == null || mSelectedNetworks.isEmpty()) return;
+        int totalNetworks = mSelectedNetworks.size();
+        AtomicInteger resCount = new AtomicInteger();
```

Comment:
because cannot use local variable as it is being used within the callback and don't wanna use a global variable to avoid logical complexity so this is a decent solution.